packages/server/lambdas/alfresco/delete-folder.ts

```diff
@@ -0,0 +1,78 @@
+import { CategoryDataBase } from '@prisma/client';
+import { ALBEvent, ALBResult } from 'aws-lambda';
+import { findEndpoint, getAlfrescoOptions, getAlfrescoUrlBase } from '../../utils/alfresco';
+import { getRataExtraLambdaError, RataExtraLambdaError } from '../../utils/errors';
+import { log, auditLog } from '../../utils/logger';
+import { getUser, validateReadUser, validateWriteUser } from '../../utils/userService';
+import { DatabaseClient } from '../database/client';
+import { folderDeleteRequestBuilder } from './fileRequestBuilder';
+import fetch from 'node-fetch';
+import { RequestInit } from 'node-fetch';
+import { AlfrescoResponse } from './fileRequestBuilder/types';
+import { deleteComponent } from '../database/components/delete-node-component';
+
+const database = await DatabaseClient.build();
+
+let fileEndpointsCache: Array<CategoryDataBase> = [];
+
+const deleteFolder = async (options: RequestInit, nodeId: string): Promise<AlfrescoResponse | undefined> => {
+  const alfrescoCoreAPIUrl = `${getAlfrescoUrlBase()}/alfresco/versions/1`;
+  const url = `${alfrescoCoreAPIUrl}/nodes/${nodeId}`;
+  try {
+    const res = await fetch(url, options);
+    const result = (await res.json()) as AlfrescoResponse;
+    return result;
+  } catch (err) {
+    console.error('error:' + err);
+  }
+};
+
+/**
+ * Delete Alfresco node (in this case folder). Example: /api/alfresco/file/linjakaaviot/FOO-123-AAA
+ * @param {ALBEvent} event
+ * @param {{string}} event.path Path should include category and node id of the folder to delete
+ * @returns  {Promise<ALBResult>} JSON stringified object of deleted folder
+ */
+export async function handleRequest(event: ALBEvent): Promise<ALBResult | undefined> {
+  try {
+    const paths = event.path.split('/');
+    const nodeId = paths.at(-1);
+    const category = paths.at(-2);
+
+    const user = await getUser(event);
+    log.info(user, `Deleting folder from ${category}`);
+    validateReadUser(user);
+
+    if (!category) {
+      throw new RataExtraLambdaError('Category missing from path', 400);
+    }
+    if (!nodeId) {
+      throw new RataExtraLambdaError('NodeId missing from path', 400);
+    }
+    if (!fileEndpointsCache.length) {
+      fileEndpointsCache = await database.categoryDataBase.findMany();
+    }
+    const categoryData = findEndpoint(category, fileEndpointsCache);
+    if (!categoryData) {
+      throw new RataExtraLambdaError('Category not found', 404);
+    }
+
+    const writeRole = categoryData.writeRights;
+    validateWriteUser(user, writeRole);
+
+    const headers = (await getAlfrescoOptions(user.uid)).headers;
+    const requestOptions = folderDeleteRequestBuilder(headers) as RequestInit;
+
+    const alfrescoResult = await deleteFolder(requestOptions, nodeId);
+    await deleteComponent(nodeId);
```

Comment:
Should we have conditional check if `deleteFolder` succeeded in Alfresco, so can continue with `deleteComponent` in database? In case, delete in Alfresco fails could lead to data corruption.

Reply:
This one is now fixed